src/server/server_family.cc

```diff
@@ -1976,6 +2064,41 @@ void ServerFamily::ReplicaOf(CmdArgList args, ConnectionContext* cntx) {
   }
 }
 
+void ServerFamily::ReplicaOf(CmdArgList args, ConnectionContext* cntx) {
+  ReplicaOfInternal(args, cntx, ShouldFlushDb::kFlush, ActionOnConnectionFail::kReturnOnError);
+}
+
+bool ServerFamily::Replicate(string& host, string& port) {
+  io::StringSink sink;
+  ConnectionContext ctxt{&sink, nullptr};
+
+  vector<MutableSlice> vargs{
+      {host.data(), host.size()},
+      {port.data(), port.size()},
+  };
+
+  CmdArgList args{vargs.data(), vargs.size()};
+
+  // we don't flush the database as the context is null
+  // (and also because there is nothing to flush)
+  ReplicaOfInternal(args, &ctxt, ShouldFlushDb::kDontFlush,
+                    ActionOnConnectionFail::kContinueReplication);
+
+  // Check whether replication succeeded
+
+  using namespace facade::constants;
+  string_view sv = sink.str();
+  if (absl::StartsWith(sv, kSimplePref)) {
+    VLOG(1) << "Replication success!";  // if ReplicaOfInternal recieves kContinueReplication then
+                                        // theoretically we should never fail
+    return true;
+  } else {
+    VLOG(1) << "Replication failure!";
```

Comment:
but you always send ok. can we somehow reach this branch?

Reply:
yes, if we decide someday to change the call to `ReplicaOfInternal` to have `kReturnOnError` then we can reach this branch. Not right now, though, so I'll remove this